src/DependencyInjection/Configuration.php

```diff
@@ -22,9 +23,23 @@ public function getConfigTreeBuilder(): TreeBuilder
 
         $rootNode
             ->children()
-            ->enumNode('driver')
-            ->values(['doctrine', 'elasticsearch'])
-            ->defaultValue('doctrine')
+                ->enumNode('driver')
+                    ->values(['doctrine', 'elasticsearch'])
+                    ->defaultValue('doctrine')
+                ->end()
+                ->scalarNode('templates_dir')
+                    ->defaultValue('catalog')
+                    ->cannotBeEmpty()
+                    ->validate()
+                        ->always(function ($value) {
+                            if (!is_string($value)) {
```

Comment:
Also i'd validate if `is_dir`

Reply:
Imo - yes, as configuration says, that's directory, so it should be valid directory (if you're not creating this within your logic)

Reply:
Added + add info to docs